Feature/sdl 0188 interior vehicle data resumption (#225)
smartdevicelink/sdl_hmi_integration_guidelines: docs/RC/GetInteriorVehicleData/index.md

The HMI should return interior vehicle data that corresponds to the requested module (`moduleId` + `moduleType`).

> During data resumption process SDL sends GetInteriorVehicleData(subscribe=true) request to HMI and stores data received from HMI in cache.

Reviewer: ```suggestion
During the data resumption process, SDL sends GetInteriorVehicleData(subscribe=true) request to the HMI and stores data received from the HMI in cache.
```

HMI must provide optional `moduleId` param  in GetInteriorVehicleData_response.
HMI must return in GetInteriorVehicleData_response the current value of the display mode used in HMI if `moduleType = HMI_SETTINGS` .

!!! NOTE

> If during resumption HMI responds with error to GetInteriorVehicleData request SDL reverts already subscribed data and fails resumption, removing information about this subscription.  

Reviewer 2: If during resumption HMI responds with error to GetInteriorVehicleData request or responds with SUCCESS to GetInteriorVehicleData  but with parameter isSubscribed=false, SDL reverts already subscribed data and fails resumption for related application(s), removing information about this subscription.

Author: thank you, done in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/225/commits/9074c7d61dea3f16ee213393369743b367f7b50d

> In the case after reverting the subscription, there is no application subscribed to a certain module type, SDL sends GetInteriorVehicleData (subscribe=false) to HMI and clears cache for this module type. 

Reviewer: ```suggestion
After reverting the subscription, if there is no application subscribed to a certain module type, SDL sends GetInteriorVehicleData (subscribe=false) to the HMI and clears the cache for this module type.
```
